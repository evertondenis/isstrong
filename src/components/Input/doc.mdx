---
name: Input
menu: Components
---

import { Playground, PropsTable } from 'docz'
import Input from './'

# Input

<PropsTable of={Input} />

## Basic usage

<Playground>
  <Input label="Input text" onChange={() => null} />
  <Input label="Input password" type="password" onChange={() => null} />
</Playground>
