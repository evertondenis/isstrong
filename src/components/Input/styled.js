import styled from 'styled-components'

export const StyledGroup = styled.div`
  label {
    color: #696d8c;
  }

  input {
    margin: 10px 0 25px;
    padding: 10px;
    width: 100%;
    background: #fff;
    border: 1px solid #b6b9d0;
    box-sizing: border-box;
    box-shadow: inset 0px 3px 3px rgba(0, 0, 0, 0.05);
    font-size: 1rem;
    color: #696d8c;
    outline: none;
    transition: all .2s linear;

    &:focus {
      outline: none;
    }
  }
`

export const StyledInput = styled.input`
  margin: 10px 0 25px;
  padding: 10px;
  width: 100%;
  background: #fff;
  border: 1px solid #b6b9d0;
  box-sizing: border-box;
  box-shadow: inset 0px 3px 3px rgba(0, 0, 0, 0.05);
  font-size: 1rem;
  color: #696d8c;
  outline: none;
  transition: all .2s linear;

  &:focus {
    outline: none;
  }
`
