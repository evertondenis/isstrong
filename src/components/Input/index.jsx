import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { StyledGroup } from './styled'

class Input extends Component {
  render() {
    const { type, focus, onChange, className, label, placeholder, required } = this.props
    return (
      <StyledGroup>
        {label && <label>{label}</label>}
        <input
          className={className}
          ref="inputRef"
          type={type}
          onChange={() => onChange()}
          placeholder={placeholder}
          autoFocus={focus}
          required={required}
        />
      </StyledGroup>
    )
  }
}

Input.defaultProps = {
  type: 'text',
  className: '',
  label: '',
  placeholder: '',
  focus: false,
  required: false
}

Input.propTypes = {
  type: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  focus: PropTypes.bool,
  required: PropTypes.bool
}

export default Input
