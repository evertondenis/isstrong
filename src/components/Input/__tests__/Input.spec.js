import React from 'react'
import { shallow, mount } from 'enzyme'
import 'jest-styled-components'
import Input from '../'
import 'setupTest'

describe('Input.', () => {
  it('Should be render <Input />.', () => {
    const onChange = jest.fn()
    const component = shallow(
      <Input
        className="class"
        label="E-mail"
        ref="inputRefEmail"
        onChange={onChange}
        required
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('should be onChange', () => {
    const onChange = jest.fn()
    const component = mount(
      <Input
        className="class"
        label="E-mail"
        ref="inputRefEmail"
        onChange={onChange}
        required
      />
    )

    component.find('input').simulate('change')
    expect(onChange).toHaveBeenCalled()
  })
})
