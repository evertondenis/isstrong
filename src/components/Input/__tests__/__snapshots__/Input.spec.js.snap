// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input. Should be render <Input />. 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    className="class"
    focus={false}
    label="E-mail"
    onChange={[Function]}
    placeholder=""
    required={true}
    type="text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label>
          E-mail
</label>,
        <input
          autoFocus={false}
          className="class"
          onChange={[Function]}
          placeholder=""
          required={true}
          type="text"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "E-mail",
        },
        "ref": null,
        "rendered": "E-mail",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoFocus": false,
          "className": "class",
          "onChange": [Function],
          "placeholder": "",
          "required": true,
          "type": "text",
        },
        "ref": "inputRef",
        "rendered": null,
        "type": "input",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label>
            E-mail
</label>,
          <input
            autoFocus={false}
            className="class"
            onChange={[Function]}
            placeholder=""
            required={true}
            type="text"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "E-mail",
          },
          "ref": null,
          "rendered": "E-mail",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": false,
            "className": "class",
            "onChange": [Function],
            "placeholder": "",
            "required": true,
            "type": "text",
          },
          "ref": "inputRef",
          "rendered": null,
          "type": "input",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
